devicetypes/smartthings/zwave-motion-sensor.src/zwave-motion-sensor.groovy

```diff
@@ -61,18 +62,44 @@ metadata {
 		main "motion"
 		details(["motion", "battery", "tamper"])
 	}
+		preferences {
+			section{
+				input(tittle: "Everspring Motion Sensor settings",
+					description: "The Configuration parameter that can be used to adjust the interval of being re-triggered after the detector has been triggered as Configuration Parameter #4.  No response will be made during this interval if a movement is presented.  The time interval can be set between 5 secs to 3600 secs.",
+ 					displayDuringSetup: false,
+					type: "paragraph",
+					element: "paragraph"
+					)
+
+				input("reTriggerValue", "number",
+					title: "Re-Trigger Interval",
+					description: "5 - 3600",
+					defaultValue: "180",
+					range: "5..3600",
+					required: "false"
+					)
+			}
+		}
 }
 
 def installed() {
 // Device wakes up every 4 hours, this interval allows us to miss one wakeup notification before marking offline
 	sendEvent(name: "checkInterval", value: 8 * 60 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID])
 	sendEvent(name: "tamper", value: "clear", displayed: false)
+	state.reTriggerMap = [:]
+	if(isEverspring()) {
+		state.reTriggerMap = reTriggerValue ? reTriggerValue : false
+	}
+	
 	response(initialPoll())
 }
 
 def updated() {
 // Device wakes up every 4 hours, this interval allows us to miss one wakeup notification before marking offline
 	sendEvent(name: "checkInterval", value: 8 * 60 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID])
+	if(isEverspring()) {
+		state.reTriggerMap = reTriggerValue ? reTriggerValue : false
```

Comment:
The values you're putting into this "map" are inconsistent. In addition to calling it a "map", here you're putting in either an integer or a boolean, while below you put a one-value array of `[0]`.

Reply:
So I should delete this map and leave only value on 203 line?

Reply:
So your intention is to store a value when the setting is changed, send that value on the next device wakeup, and then clear it, and not send the setting on next wakeup if you've already sent it, right?

Here's what your code should be doing:
1. When the setting is changed, update `state.newTriggerValue`
2. When you receive a wakeup notification, check the value of `state.newTriggerValue`
3. If you have a non-`null` value, send the configuration command and pass in `state.newTriggerValue`
4. Set `state.newTriggerValue` to `null`